src/aap_eda/api/views/decision_environment.py

```diff
@@ -117,3 +109,18 @@ def retrieve(self, request, pk):
                 decision_environment.data
             ).data
         )
+
+    @extend_schema(
+        description="Delete a decision environment by id",
+        responses=status.HTTP_204_NO_CONTENT,
+    )
+    def perform_destroy(self, instance):
+        activations = models.Activation.objects.filter(
+            decision_environment_id=instance.id
+        ).values_list("id", flat=True)
+        if activations:
+            raise api_exc.Conflict(
```

Comment:
To me this is the right approach, it should be done also for extra_Vars, credentials and token. But it needs to be discussed, last conversation with @benthomasson he didn't want to do this. Even if we go ahead with this, UI should be aligned to capture these cases.